it/controllers/benefits/medical/MedicalOrDentalBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,583 @@
+              _.copy(benefits = fullBenefits.flatMap(_.benefits.map(_.copy(medicalInsurance = None))))
+            ))))), nino, taxYearEOY)
+            insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true,
+              Some(benefits(fullMedicalChildcareEducationModel.copy(medicalInsurance = None))))), userRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlPost(url(taxYearEOY), form, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an BAD_REQUEST status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+          titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(user.commonExpectedResults.expectedCaption(taxYearEOY))
+          buttonCheck(user.commonExpectedResults.continueButtonText, continueButtonSelector)
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraph, paragraphTextSelector(2))
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraphforForm, paragraphTextSelector(3))
+          errorSummaryCheck(user.specificExpectedResults.get.expectedNoEntryErrorMessage, amountField)
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("", amountField)
+          formPostLinkCheck(continueLink(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+
+        }
+
+      }
+    }
+    "redirect to another page when request is valid" should {
+
+      val user = UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedAgentEN))
+
+      lazy val form: Map[String, String] = Map(AmountForm.amount -> "97")
+
+      "Redirect to cya page when all is valid in previous data flow" which {
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          userDataStub(userData(fullEmploymentsModel(hmrcEmployment = Seq(employmentDetailsAndBenefits(fullBenefits)))), nino, taxYearEOY)
+          insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true,
+            Some(benefits(fullMedicalChildcareEducationModel)))), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(url(taxYearEOY), form, follow = false, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+        }
+
+      }
+
+      "Redirect to cya page when all is valid in new data flow" which {
+        //should change to childcare later
```

Comment:
id change this to a TODO instead of just a comment so that its easier to identify in the code